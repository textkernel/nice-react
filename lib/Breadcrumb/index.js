export { default as BreadcrumbGroup } from './BreadcrumbGroup';
export { default as BreadcrumbItem } from './BreadcrumbItem';