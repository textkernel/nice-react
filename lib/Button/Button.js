import React, { Component } from 'react';
import PropTypes from 'prop-types';
import niceProps from '../props';

export default class Button extends Component {
  handleClick(e) {
    e.preventDefault();
    const { onClick } = this.props;
    if (!onClick) {
      return false;
    }
    onClick();
    return true;
  }

  render() {
    const { children, context, size } = this.props;
    const classes = ['nice-btn', context, size].filter(c => c && c.length > 0);

    return (
      <button type="button" onClick={ e => this.handleClick(e) } className={classes.join(' ')}>
        {children}
      </button>
    );
  }
}

Button.propTypes = {
  context: PropTypes.oneOf(niceProps.buttonContexts),
  size: PropTypes.oneOf(niceProps.buttonSizes),
  children: PropTypes.node,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  context: 'primary',
  size: 'medium',
  children: null,
  onClick: null
};
