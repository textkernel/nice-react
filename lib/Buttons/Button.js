import React, { Component } from 'react';
import PropTypes from 'prop-types';
import props from '../props';

export default class Button extends Component {
  handleClick(e) {
    e.preventDefault();
    this.props.onClick();
  }

  render() {
    const { children, icon, alternative } = this.props;
    const iconClass = icon ? `icon ${icon}` : null;

    const regularButton = () => {
      const { context, size } = this.props;
      const baseClass = 'nice-btn';
      return [baseClass, context, size, iconClass];
    };

    const alternativeButton = () => {
      const baseClass = 'nice-btn-alt';
      return [baseClass, iconClass];
    };

    const butonClasses = alternative ? alternativeButton : regularButton;
    const classes = butonClasses().filter(c => c && c.length > 0);

    return (
      <a href='#' onClick={e => this.handleClick(e)} className={classes.join(' ')}>
        {children}
      </a>
    );
  }
}

Button.propTypes = {
  context: PropTypes.oneOf(props.buttonContexts),
  size: PropTypes.oneOf(props.buttonSizes),
  children: PropTypes.node,
  onClick: PropTypes.func,
  icon: PropTypes.string,
  alternative: PropTypes.bool
};

Button.defaultProps = {
  context: 'default',
  size: 'medium',
  children: null,
  onClick: null,
  icon: null,
  alternative: false
};