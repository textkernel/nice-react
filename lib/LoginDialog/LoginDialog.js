import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import niceProps from '../props';
import ProductIcon from '../ProductIcon';

export default class LoginDialog extends Component {
    static propTypes = {
        brand: PropTypes.string.isRequired,
        children: PropTypes.node,
        product: PropTypes.oneOf(niceProps.products),
        intro: PropTypes.oneOfType([
            PropTypes.node,
            PropTypes.string
        ]),
        loading: PropTypes.bool,
        version: PropTypes.string,
        noAttribution: PropTypes.bool
    }

    static defaultProps = {
        brand: niceProps.products[0],
        children: null,
        product: niceProps.products[0],
        intro: null,
        loading: false,
        version: null,
        noAttribution: false
    }

    renderProductLogo() {
        const { product } = this.props;

        if (!product) {
            return null;
        }
        return (<div
            className="nice-login-intro-logo"
        >
            <ProductIcon
                product={ product }
            />
        </div>);
    }

    renderIntro() {
        const { intro, loading } = this.props;

        if (!intro) {
            return null;
        }

        return (
            <div className="nice-login-intro">
                {
                    this.renderProductLogo()
                }
                <div className="nice-login-intro-content">
                    { intro }
                </div>
                <div className={ classNames('nice-login-loading', {
                    shown: loading
                }) } />
            </div>
        );
    }

    renderAttribution() {
        const { noAttribution } = this.props;

        if (noAttribution) {
            return null;
        }
        return (<span
            className="text-brand pull-right"
        >
            by textkernel
        </span>);
    }

    renderVersion() {
        const { version, noAttribution } = this.props;

        if (!version && noAttribution) {
            return null;
        }

        return (
            <div className="nice-login-version clearfix">
                { version }
                {
                    this.renderAttribution()
                }
            </div>
        );
    }

    render() {
        const { children, brand } = this.props;

        return (
            <div className="nice-login">
                <div className="nice-login-brand">
                    <span className="text-brand">
                        { brand }
                    </span>
                </div>
                { this.renderIntro() }
                { children }
                { this.renderVersion() }
            </div>
        );
    }
}
