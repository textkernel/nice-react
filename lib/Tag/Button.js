import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const Button = ({ icon, className, onClick, ...rest }) => {
    const handleClick = (event) => {
        event.stopPropagation();

        if (onClick) {
            onClick(event);
        }
    };

    return (
        <button
            className={classnames('nice-tag-btn', className)}
            onClick={handleClick}
            {...rest}
        >
            <i className={`icon ${icon}`} />
        </button>
    );
};


Button.propTypes = {
    icon: PropTypes.string.isRequired,
    className: PropTypes.string,
    onClick: PropTypes.func,
};


Button.defaultProps = {
    className: '',
    onClick: null,
};


export default Button;