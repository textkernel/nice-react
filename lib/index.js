import '@textkernel/nice/dist/nice.min.css';

export { default as Panel } from './Panel';
export { Button, ButtonGroup } from './Buttons';
export { default as Progress } from './Progress';
export { default as Loading } from './Loading';
export { BreadcrumbGroup, BreadcrumbItem } from './Breadcrumb';