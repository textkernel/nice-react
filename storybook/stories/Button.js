import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Button } from 'nice-react';

storiesOf('Button', module)
  .add('Default', () => (
    <Button onClick={() => alert('Default')}>Default</Button>
  ))
  .add('Primary', () => (
    <Button context="primary">Primary</Button>
  ))
  .add('Large', () => (
    <Button size="large">Large</Button>
  ))
  .add('Small', () => (
    <Button size="small">Small</Button>
  ));;