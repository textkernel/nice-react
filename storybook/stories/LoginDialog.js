import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobsOptions, withKnobs, boolean, select, text } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import props from '../../src/props';

import { Button, LoginDialog } from 'nice-react';

const stories = storiesOf('Login Dialog', module);

stories.addDecorator((story, context) => withInfo('')(story)(context));
stories.addDecorator(withKnobs);

stories.add('Default', () => {
    const brand = text('Brand', 'textkernel');
    const product = select('Product', props.products);
    const intro = text('Intro text', 'Welcome to Textkernel. Please log in using your credentials to access your account.')
    const version = text('Version', 'Nice! React v1.2.3');
    const loading = boolean('Loading', false);
    const attribution = boolean('Show attribution', true);

    return (<LoginDialog
        brand={ brand }
        product={ product }
        intro={ intro }
        version={ version }
        loading={ loading }
        noAttribution={ !attribution }
    >
        <form onSubmit={ e => {
            e.preventDefault();
        }}>
            <div className="nice-form-group">
                <label htmlFor="nice-login-username">Username</label>
                <input type="email" placeholder="Username..." className="nice-form-control" id="nice-login-username" />
            </div>
            <div className="nice-form-group">
                <label htmlFor="nice-login-account">Account</label>
                <input type="text" placeholder="Account name..." className="nice-form-control" id="nice-login-account" />
            </div>
            <div className="nice-form-group">
                <label htmlFor="nice-login-password">Password</label>
                <input type="password" placeholder="Password..." className="nice-form-control" id="nice-login-password" />
            </div>
            <div className="nice-form-group">
                <input type="checkbox" className="nice-toggle" id="login-remember" />
                <label className="nice-toggle-btn off-muted" htmlFor="login-remember">
                    Remember me on this device
                </label>
            </div>
            <div className="nice-form-group">
                <Button context="link" submit style={{
                    paddingLeft: 0,
                    paddingRight: 0
                }}>
                    <i className="icon fa-question" /> Lost your credentials?
                </Button>
                <Button context="primary" submit pullRight>
                    Login
                </Button>
            </div>
        </form>
    </LoginDialog>);
});
