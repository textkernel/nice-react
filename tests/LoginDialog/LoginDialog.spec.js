import React from 'react';
import LoginDialog from '../../lib/LoginDialog';
import { Button } from '../../lib/Button';

describe('<LoginDialog />', () => {
    it('should render a default LoginDialog', () => {
        const wrapper = shallow(<LoginDialog
            brand="textkernel"
            product="textkernel"
            intro="Welcome to Textkernel. Please log in using your credentials to access your account."
            version="Nice v1.2.3"
        >
            <form onSubmit={ e => {
                e.preventDefault();
            }}>
                <div className="nice-form-group">
                    <label htmlFor="nice-login-username">Username</label>
                    <input type="email" placeholder="Username..." className="nice-form-control" id="nice-login-username" />
                </div>
                <div className="nice-form-group">
                    <label htmlFor="nice-login-account">Account</label>
                    <input type="text" placeholder="Account name..." className="nice-form-control" id="nice-login-account" />
                </div>
                <div className="nice-form-group">
                    <label htmlFor="nice-login-password">Password</label>
                    <input type="password" placeholder="Password..." className="nice-form-control" id="nice-login-password" />
                </div>
                <div className="nice-form-group">
                    <input type="checkbox" className="nice-toggle" id="login-remember" />
                    <label className="nice-toggle-btn off-muted" htmlFor="login-remember">
                        Remember me on this device
                    </label>
                </div>
                <div className="nice-form-group">
                    <Button context="link" submit style={{
                        paddingLeft: 0,
                        paddingRight: 0
                    }}>
                        <i className="icon fa-question" /> Lost your credentials?
                    </Button>
                    <Button context="primary" submit className="pull-right">
                        Login
                    </Button>
                </div>
            </form>
        </LoginDialog>);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render a loading LoginDialog without attribution / version', () => {
        const wrapper = shallow(<LoginDialog
            brand="textkernel"
            product="textkernel"
            intro="Welcome to Textkernel. Please log in using your credentials to access your account."
            version="Nice v1.2.3"
            noAttribution
            loading
        >
            <form onSubmit={ e => {
                e.preventDefault();
            }}>
                <div className="nice-form-group">
                    <label htmlFor="nice-login-username">Username</label>
                    <input type="email" placeholder="Username..." className="nice-form-control" id="nice-login-username" />
                </div>
                <div className="nice-form-group">
                    <label htmlFor="nice-login-account">Account</label>
                    <input type="text" placeholder="Account name..." className="nice-form-control" id="nice-login-account" />
                </div>
                <div className="nice-form-group">
                    <label htmlFor="nice-login-password">Password</label>
                    <input type="password" placeholder="Password..." className="nice-form-control" id="nice-login-password" />
                </div>
                <div className="nice-form-group">
                    <input type="checkbox" className="nice-toggle" id="login-remember" />
                    <label className="nice-toggle-btn off-muted" htmlFor="login-remember">
                        Remember me on this device
                    </label>
                </div>
                <div className="nice-form-group">
                    <Button context="link" submit style={{
                        paddingLeft: 0,
                        paddingRight: 0
                    }}>
                        <i className="icon fa-question" /> Lost your credentials?
                    </Button>
                    <Button context="primary" submit className="pull-right">
                        Login
                    </Button>
                </div>
            </form>
        </LoginDialog>);

        expect(wrapper).toMatchSnapshot();
    });
});
