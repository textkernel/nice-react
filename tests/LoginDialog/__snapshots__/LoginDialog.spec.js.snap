// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginDialog /> should render a default LoginDialog with custom className/style 1`] = `
<div
  className="nice-login my-class"
  style={
    Object {
      "top": 0,
    }
  }
>
  <div
    className="nice-login-brand"
  >
    <span
      className="text-brand"
    >
      textkernel
    </span>
  </div>
  <div
    className="nice-login-intro"
  >
    <div
      className="nice-login-intro-logo"
    >
      <ProductIcon
        className=""
        product="textkernel"
        style={null}
      />
    </div>
    <div
      className="nice-login-intro-content"
    >
      Welcome to Textkernel. Please log in using your credentials to access your account.
    </div>
    <div
      className="nice-login-loading"
    />
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="nice-form-group"
    >
      <label
        htmlFor="nice-login-username"
      >
        Username
      </label>
      <input
        className="nice-form-control"
        id="nice-login-username"
        placeholder="Username..."
        type="email"
      />
    </div>
    <div
      className="nice-form-group"
    >
      <label
        htmlFor="nice-login-account"
      >
        Account
      </label>
      <input
        className="nice-form-control"
        id="nice-login-account"
        placeholder="Account name..."
        type="text"
      />
    </div>
    <div
      className="nice-form-group"
    >
      <label
        htmlFor="nice-login-password"
      >
        Password
      </label>
      <input
        className="nice-form-control"
        id="nice-login-password"
        placeholder="Password..."
        type="password"
      />
    </div>
    <div
      className="nice-form-group"
    >
      <input
        className="nice-toggle"
        id="login-remember"
        type="checkbox"
      />
      <label
        className="nice-toggle-btn off-muted"
        htmlFor="login-remember"
      >
        Remember me on this device
      </label>
    </div>
    <div
      className="nice-form-group"
    >
      <Button
        alternative={false}
        className=""
        context="link"
        disabled={false}
        href={null}
        icon={null}
        onClick={null}
        pullRight={false}
        size="medium"
        style={
          Object {
            "paddingLeft": 0,
            "paddingRight": 0,
          }
        }
        submit={true}
      >
        <i
          className="icon fa-question"
        />
         Lost your credentials?
      </Button>
      <Button
        alternative={false}
        className="pull-right"
        context="primary"
        disabled={false}
        href={null}
        icon={null}
        onClick={null}
        pullRight={false}
        size="medium"
        style={null}
        submit={true}
      >
        Login
      </Button>
    </div>
  </form>
  <div
    className="nice-login-version clearfix"
  >
    Nice v1.2.3
    <span
      className="text-brand pull-right"
    >
      by textkernel
    </span>
  </div>
</div>
`;

exports[`<LoginDialog /> should render a loading LoginDialog without attribution / version 1`] = `
<div
  className="nice-login"
  style={Object {}}
>
  <div
    className="nice-login-brand"
  >
    <span
      className="text-brand"
    >
      textkernel
    </span>
  </div>
  <div
    className="nice-login-intro"
  >
    <div
      className="nice-login-intro-logo"
    >
      <ProductIcon
        className=""
        product="textkernel"
        style={null}
      />
    </div>
    <div
      className="nice-login-intro-content"
    >
      Welcome to Textkernel. Please log in using your credentials to access your account.
    </div>
    <div
      className="nice-login-loading shown"
    />
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="nice-form-group"
    >
      <label
        htmlFor="nice-login-username"
      >
        Username
      </label>
      <input
        className="nice-form-control"
        id="nice-login-username"
        placeholder="Username..."
        type="email"
      />
    </div>
    <div
      className="nice-form-group"
    >
      <label
        htmlFor="nice-login-account"
      >
        Account
      </label>
      <input
        className="nice-form-control"
        id="nice-login-account"
        placeholder="Account name..."
        type="text"
      />
    </div>
    <div
      className="nice-form-group"
    >
      <label
        htmlFor="nice-login-password"
      >
        Password
      </label>
      <input
        className="nice-form-control"
        id="nice-login-password"
        placeholder="Password..."
        type="password"
      />
    </div>
    <div
      className="nice-form-group"
    >
      <input
        className="nice-toggle"
        id="login-remember"
        type="checkbox"
      />
      <label
        className="nice-toggle-btn off-muted"
        htmlFor="login-remember"
      >
        Remember me on this device
      </label>
    </div>
    <div
      className="nice-form-group"
    >
      <Button
        alternative={false}
        className=""
        context="link"
        disabled={false}
        href={null}
        icon={null}
        onClick={null}
        pullRight={false}
        size="medium"
        style={
          Object {
            "paddingLeft": 0,
            "paddingRight": 0,
          }
        }
        submit={true}
      >
        <i
          className="icon fa-question"
        />
         Lost your credentials?
      </Button>
      <Button
        alternative={false}
        className="pull-right"
        context="primary"
        disabled={false}
        href={null}
        icon={null}
        onClick={null}
        pullRight={false}
        size="medium"
        style={null}
        submit={true}
      >
        Login
      </Button>
    </div>
  </form>
  <div
    className="nice-login-version clearfix"
  >
    Nice v1.2.3
  </div>
</div>
`;
